sqlmesh/core/context_diff.py

```diff
@@ -475,29 +483,41 @@ def _build_requirements(
     provided_requirements: t.Dict[str, str],
     excluded_requirements: t.Set[str],
     snapshots: t.Collection[Snapshot],
+    infer_python_dependencies: bool = True,
 ) -> t.Dict[str, str]:
     requirements = {
         k: v for k, v in provided_requirements.items() if k not in excluded_requirements
     }
+
+    if not infer_python_dependencies:
+        return requirements
+
     distributions = metadata.packages_distributions()
 
     for snapshot in snapshots:
-        if snapshot.is_model:
-            for executable in snapshot.model.python_env.values():
-                if executable.kind == "import":
-                    try:
-                        start = "from " if executable.payload.startswith("from ") else "import "
-                        lib = executable.payload.split(start)[1].split()[0].split(".")[0]
-                        if lib in distributions:
-                            for dist in distributions[lib]:
-                                if (
-                                    dist not in requirements
-                                    and dist not in IGNORED_PACKAGES
-                                    and dist not in excluded_requirements
-                                ):
-                                    requirements[dist] = metadata.version(dist)
-                    except metadata.PackageNotFoundError:
-                        from sqlmesh.core.console import get_console
-
-                        get_console().log_warning(f"Failed to find package for {lib}.")
+        if not snapshot.is_model:
+            continue
```

Comment:
I went ahead and de-nested this section a bit as well. Didn't change anything besides negating conditions and short-circuiting with `continue`.